drivers/backup/helper/backup_helper.go

```diff
@@ -0,0 +1,90 @@
+package helper
+
+import (
+	"fmt"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/pkg/log"
+	"github.com/portworx/torpedo/pkg/testrailuttils"
+	"github.com/portworx/torpedo/tests"
+	"gopkg.in/natefinch/lumberjack.v2"
+	"strconv"
+	"strings"
+)
+
+var (
+	pxBackupTorpedoTestInfoMap = make(map[int]*PxBackupTorpedoTestInfo, 0)
+)
+
+// PxBackupTorpedoTestInfo holds information of a particular test
+type PxBackupTorpedoTestInfo struct {
+	testName          string
+	testDescription   string
+	testAuthor        string
+	testRailID        int
+	testRunIdForSuite int
+	testTags          map[string]string
+	testLogger        *lumberjack.Logger
+}
+
+// StartPxBackupTorpedoTest creates a logger, configures the Aetos Dashboard for the specified test, and initializes controllers
+func StartPxBackupTorpedoTest(testRailId int, testName string, testDescription string, testAuthor string, apps []string, tags ...map[string]string) error {
+	if testRailId != 0 {
+		if pxBackupTorpedoTestInfo, ok := pxBackupTorpedoTestInfoMap[testRailId]; ok {
+			err := fmt.Errorf("the test [%s] shares the same TestRail id as [%s] and has already been executed", testName, pxBackupTorpedoTestInfo.testName)
+			return err
+		}
+	}
+	testTags := map[string]string{
+		"author": testAuthor,
+		"apps":   strings.Join(apps, ","),
+	}
+	if len(tags) > 0 {
+		invalidTagKeys := make([]string, 0)
+		for tagKey := range testTags {
+			invalidTagKeys = append(invalidTagKeys, tagKey)
+		}
+		for tagKey, tagValue := range tags[0] {
+			for _, invalidTagKey := range invalidTagKeys {
+				if tagKey == invalidTagKey {
+					err := fmt.Errorf("tags cannot have any of the keys [%s]", invalidTagKeys)
+					return err
+				}
+			}
+			testTags[tagKey] = tagValue
+		}
```

Comment:
Why is this needed?

Reply:
To prevent from overriding the author and apps keys.

Reply:
This is not required